js/piwik.js

```diff
@@ -4962,14 +5000,23 @@ if (typeof Piwik !== 'object') {
                 /**
                  * Set heartbeat (in seconds)
                  *
-                 * @param int minimumVisitLength
-                 * @param int heartBeatDelay
+                 * @param int heartBeatDelayInSeconds Defaults to 15.
                  */
-                setHeartBeatTimer: function (minimumVisitLength, heartBeatDelay) {
-                    var now = new Date();
+                enableHeartBeatTimer: function (heartBeatDelayInSeconds) {
+                    configHeartBeatDelayInSeconds = (heartBeatDelayInSeconds || 15) * 1000;
 
-                    configMinimumVisitTime = now.getTime() + minimumVisitLength * 1000;
-                    configHeartBeatTimer = heartBeatDelay * 1000;
+                    // if a tracking request has already been sent, start the heart beat timeout
+                    if (lastTrackerRequestTime !== null) {
+                        setUpHeartBeat();
+                    }
+                },
+
+                /**
+                 * Clear heartbeat.
+                 */
+                disableHeartBeatTimer: function () {
+                    heartBeatDown();
+                    configHeartBeatDelayInSeconds = null;
```

Comment:
There might be a problem. What I did was enabling heart beat timer -> works. Went to the developer console and the ping stopped as it lost focus :+1: . Then I called `_paq.push(['disableHeartBeatTimer']);` in the developer console, then I focused the window again and it started sending ping requests again although it is disabled. 

I reckon it could be nice to have a check in `heartBeatUp()` to actually setup the timer if there is really a delay specified or so. I tested with Google Chrome 43 on MacOS.

I think there is no check in `heartBeatPingIfActivityAlias` whether a `configHeartBeatDelayInSeconds` is enabled/set when coming via `window.focus`.